CVE export error "404 page not found"

Hi all,

In our environment (Harbor `v2.7.0`) we detected the issue with exporting the CVE report.

![image](https://user-images.githubusercontent.com/99802718/226391699-aac278bc-8a54-4d52-82b7-1a3aa224cf10.png)

`jobservice.log:`
```
Mar 20, 2023 @ 19:46:19.396    Mar 20 15:16:19 172.20.0.1 jobservice[25867]: 2023-03-20T14:16:19Z [ERROR] [/jobservice/job/impl/scandataexport/scan_data_export.go:130]: Export Job Id = 7.678138e+06. Error when persisting report file /var/scandata_exports/scandata_export_7.678138e+06.csv to persistent storage: http status code: 404, body: 404 page not found    
Mar 20, 2023 @ 19:45:48.329    Mar 20 15:15:48 172.20.0.1 jobservice[25867]: 2023-03-20T14:15:48Z [ERROR] [/jobservice/job/impl/scandataexport/scan_data_export.go:130]: Export Job Id = 7.678137e+06. Error when persisting report file /var/scandata_exports/scandata_export_7.678137e+06.csv to persistent storage: http status code: 404, body: 404 page not found    
Mar 20, 2023 @ 19:45:25.889    Mar 20 15:15:25 172.20.0.1 jobservice[25867]: 2023-03-20T14:15:25Z [ERROR] [/jobservice/job/impl/scandataexport/scan_data_export.go:130]: Export Job Id = 7.678135e+06. Error when persisting report file /var/scandata_exports/scandata_export_7.678135e+06.csv to persistent storage: http status code: 404, body: 404 page not found
```

We already checked the permissions in the filesystem and the Docker configuration and we didn't saw any "strange" configuration.

Does anyone have an idea, why this is not working?

P.S.: Maybe related to https://github.com/goharbor/harbor-helm/issues/1444

Thank you in advance!
<br ><br >
Alexander Barth (alexander.barth@mercedes-benz.com) on behalf of Mercedes-Benz Tech Innovation GmbH, [Provider Information](https://github.com/mercedes-benz/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

--- a/src/controller/scandataexport/execution.go
+++ b/src/controller/scandataexport/execution.go
@@ -128,8 +128,8 @@ func (c *controller) Start(ctx context.Context, request export.Request) (executi
 
 	// create a job object and fill with metadata and parameters
 	params := make(map[string]interface{})
-	params["JobId"] = id
-	params["Request"] = request
+	params[export.JobID] = fmt.Sprintf("%d", id)
+	params[export.JobRequest] = request
 	params[export.JobModeKey] = export.JobModeExport
 
 	j := &task.Job{

--- a/src/jobservice/job/impl/scandataexport/scan_data_export.go
+++ b/src/jobservice/job/impl/scandataexport/scan_data_export.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
 
 	"github.com/gocarina/gocsv"
@@ -78,7 +79,7 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 	logger := ctx.GetLogger()
 	logger.Infof("Scan data export job started in mode : %v", mode)
 	sde.init()
-	fileName := fmt.Sprintf("%s/scandata_export_%v.csv", sde.scanDataExportDirPath, params["JobId"])
+	fileName := fmt.Sprintf("%s/scandata_export_%s.csv", sde.scanDataExportDirPath, params[export.JobID])
 
 	// ensure that CSV files are cleared post the completion of the Run.
 	defer sde.cleanupCsvFile(ctx, fileName, params)
@@ -93,12 +94,12 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 		logger.Errorf("Error when calculating checksum for generated file: %v", err)
 		return err
 	}
-	logger.Infof("Export Job Id = %v, FileName = %s, Hash = %v", params["JobId"], fileName, hash)
+	logger.Infof("Export Job Id = %s, FileName = %s, Hash = %v", params[export.JobID], fileName, hash)
 
 	csvFile, err := os.OpenFile(fileName, os.O_RDONLY, os.ModePerm)
 	if err != nil {
 		logger.Errorf(
-			"Export Job Id = %v. Error when moving report file %s to persistent storage: %v", params["JobId"], fileName, err)
+			"Export Job Id = %s. Error when moving report file %s to persistent storage: %v", params[export.JobID], fileName, err)
 		return err
 	}
 	baseFileName := filepath.Base(fileName)
@@ -109,7 +110,7 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 		logger.Errorf("Error when fetching file size: %v", err)
 		return err
 	}
-	logger.Infof("Export Job Id = %v. CSV file size: %d", params["JobId"], stat.Size())
+	logger.Infof("Export Job Id = %s. CSV file size: %d", params[export.JobID], stat.Size())
 	// earlier return and update status message if the file size is 0, unnecessary to push a empty system artifact.
 	if stat.Size() == 0 {
 		extra := map[string]interface{}{
@@ -117,10 +118,10 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 		}
 		updateErr := sde.updateExecAttributes(ctx, params, extra)
 		if updateErr != nil {
-			logger.Errorf("Export Job Id = %v. Error when updating the exec extra attributes 'status_message' to 'No vulnerabilities found or matched': %v", params["JobId"], updateErr)
+			logger.Errorf("Export Job Id = %s. Error when updating the exec extra attributes 'status_message' to 'No vulnerabilities found or matched': %v", params[export.JobID], updateErr)
 		}
 
-		logger.Infof("Export Job Id = %v. Exported CSV file is empty, skip to push system artifact, exit job", params["JobId"])
+		logger.Infof("Export Job Id = %s. Exported CSV file is empty, skip to push system artifact, exit job", params[export.JobID])
 		return nil
 	}
 
@@ -128,14 +129,14 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 	artID, err := sde.sysArtifactMgr.Create(ctx.SystemContext(), &csvExportArtifactRecord, csvFile)
 	if err != nil {
 		logger.Errorf(
-			"Export Job Id = %v. Error when persisting report file %s to persistent storage: %v", params["JobId"], fileName, err)
+			"Export Job Id = %s. Error when persisting report file %s to persistent storage: %v", params[export.JobID], fileName, err)
 		return err
 	}
 
-	logger.Infof("Export Job Id = %v. Created system artifact: %v for report file %s to persistent storage: %v", params["JobId"], artID, fileName, err)
+	logger.Infof("Export Job Id = %s. Created system artifact: %v for report file %s to persistent storage: %v", params[export.JobID], artID, fileName, err)
 	err = sde.updateExecAttributes(ctx, params, map[string]interface{}{export.DigestKey: hash.String()})
 	if err != nil {
-		logger.Errorf("Export Job Id = %v. Error when updating execution record : %v", params["JobId"], err)
+		logger.Errorf("Export Job Id = %s. Error when updating execution record : %v", params[export.JobID], err)
 		return err
 	}
 	logger.Info("Scan data export job completed")
@@ -144,11 +145,16 @@ func (sde *ScanDataExport) Run(ctx job.Context, params job.Parameters) error {
 }
 
 func (sde *ScanDataExport) updateExecAttributes(ctx job.Context, params job.Parameters, attrs map[string]interface{}) error {
-	execID := int64(params["JobId"].(float64))
-	exec, err := sde.execMgr.Get(ctx.SystemContext(), execID)
 	logger := ctx.GetLogger()
+	execID, err := strconv.ParseInt(params[export.JobID].(string), 10, 64)
 	if err != nil {
-		logger.Errorf("Export Job Id = %v. Error when fetching execution record for update : %v", params["JobId"], err)
+		logger.Errorf("Export Job Id = %s. Error when parse execution id from params: %v", params[export.JobID], err)
+		return err
+	}
+
+	exec, err := sde.execMgr.Get(ctx.SystemContext(), execID)
+	if err != nil {
+		logger.Errorf("Export Job Id = %s. Error when fetching execution record for update : %v", params[export.JobID], err)
 		return err
 	}
 	// copy old extra
@@ -177,8 +183,8 @@ func (sde *ScanDataExport) writeCsvFile(ctx job.Context, params job.Parameters, 
 	var exportParams export.Params
 	var artIDGroups [][]int64
 
-	if criteira, ok := params["Request"]; ok {
-		logger.Infof("Request for export : %v", criteira)
+	if criteria, ok := params[export.JobRequest]; ok {
+		logger.Infof("Request for export : %v", criteria)
 		filterCriteria, err := sde.extractCriteria(params)
 		if err != nil {
 			return err
@@ -255,7 +261,7 @@ func (sde *ScanDataExport) writeCsvFile(ctx job.Context, params job.Parameters, 
 				logger.Infof("No more data to fetch. Exiting...")
 				break
 			}
-			logger.Infof("Export Group Id = %d, Job Id = %v, Page Number = %d, Page Size = %d Num Records = %d", groupID, params["JobId"], exportParams.PageNumber, exportParams.PageSize, len(data))
+			logger.Infof("Export Group Id = %d, Job Id = %s, Page Number = %d, Page Size = %d Num Records = %d", groupID, params[export.JobID], exportParams.PageNumber, exportParams.PageSize, len(data))
 
 			// for the first page write the CSV with the headers
 			if exportParams.PageNumber == 1 && groupID == 0 {
@@ -278,9 +284,9 @@ func (sde *ScanDataExport) writeCsvFile(ctx job.Context, params job.Parameters, 
 }
 
 func (sde *ScanDataExport) extractCriteria(params job.Parameters) (*export.Request, error) {
-	filterMap, ok := params["Request"].(map[string]interface{})
+	filterMap, ok := params[export.JobRequest].(map[string]interface{})
 	if !ok {
-		return nil, errors.Errorf("malformed criteria '%v'", params["Request"])
+		return nil, errors.Errorf("malformed criteria '%v'", params[export.JobRequest])
 	}
 	jsonData, err := json.Marshal(filterMap)
 	if err != nil {
@@ -349,12 +355,12 @@ func (sde *ScanDataExport) init() {
 func (sde *ScanDataExport) cleanupCsvFile(ctx job.Context, fileName string, params job.Parameters) {
 	logger := ctx.GetLogger()
 	if _, err := os.Stat(fileName); os.IsNotExist(err) {
-		logger.Infof("Export Job Id = %v, CSV Export File = %s does not exist. Nothing to do", params["JobId"], fileName)
+		logger.Infof("Export Job Id = %s, CSV Export File = %s does not exist. Nothing to do", params[export.JobID], fileName)
 		return
 	}
 	err := os.Remove(fileName)
 	if err != nil {
-		logger.Errorf("Export Job Id = %d, CSV Export File = %s could not deleted. Error = %v", params["JobId"], fileName, err)
+		logger.Errorf("Export Job Id = %s, CSV Export File = %s could not deleted. Error = %v", params[export.JobID], fileName, err)
 		return
 	}
 }

--- a/src/jobservice/job/impl/scandataexport/scan_data_export_test.go
+++ b/src/jobservice/job/impl/scandataexport/scan_data_export_test.go
@@ -30,8 +30,11 @@ import (
 	tasktesting "github.com/goharbor/harbor/src/testing/pkg/task"
 )
 
-const JobId = float64(100)
-const MockDigest = "mockDigest"
+const (
+	ExecID     int64 = 1000000
+	JobId            = "1000000"
+	MockDigest       = "mockDigest"
+)
 
 type ScanDataExportJobTestSuite struct {
 	htesting.Suite
@@ -80,7 +83,7 @@ func (suite *ScanDataExportJobTestSuite) TestRun() {
 	execAttrs := make(map[string]interface{})
 	execAttrs[export.JobNameAttribute] = "test-job"
 	execAttrs[export.UserNameAttribute] = "test-user"
-	mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil)
+	mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil)
 
 	params := job.Parameters{}
 	params[export.JobModeKey] = export.JobModeExport
@@ -93,7 +96,7 @@ func (suite *ScanDataExportJobTestSuite) TestRun() {
 	err := suite.job.Run(ctx, params)
 	suite.NoError(err)
 	sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-		return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+		return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 	})
 	suite.sysArtifactMgr.AssertCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
 
@@ -105,8 +108,8 @@ func (suite *ScanDataExportJobTestSuite) TestRun() {
 		_, ok := m[export.CreateTimestampKey]
 		return attrsMap[export.DigestKey] == MockDigest && ok && attrsMap[export.JobNameAttribute] == "test-job" && attrsMap[export.UserNameAttribute] == "test-user"
 	})
-	suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), extraAttrsMatcher)
-	_, err = os.Stat("/tmp/scandata_export_100.csv")
+	suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, extraAttrsMatcher)
+	_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 	suite.Truef(os.IsNotExist(err), "Expected CSV file to be deleted")
 
 }
@@ -119,7 +122,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithEmptyData() {
 	execAttrs := make(map[string]interface{})
 	execAttrs[export.JobNameAttribute] = "test-job"
 	execAttrs[export.UserNameAttribute] = "test-user"
-	mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil)
+	mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil)
 
 	params := job.Parameters{}
 	params[export.JobModeKey] = export.JobModeExport
@@ -132,7 +135,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithEmptyData() {
 	extraAttrsMatcher := testifymock.MatchedBy(func(attrsMap map[string]interface{}) bool {
 		return attrsMap["status_message"] == "No vulnerabilities found or matched" && attrsMap[export.JobNameAttribute] == "test-job" && attrsMap[export.UserNameAttribute] == "test-user"
 	})
-	suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), extraAttrsMatcher)
+	suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, extraAttrsMatcher)
 }
 
 func (suite *ScanDataExportJobTestSuite) TestRunAttributeUpdateError() {
@@ -160,7 +163,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunAttributeUpdateError() {
 	err := suite.job.Run(ctx, params)
 	suite.Error(err)
 	sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-		return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+		return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 	})
 	suite.sysArtifactMgr.AssertCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
 
@@ -172,8 +175,8 @@ func (suite *ScanDataExportJobTestSuite) TestRunAttributeUpdateError() {
 		_, ok := m[export.CreateTimestampKey]
 		return attrsMap[export.DigestKey] == MockDigest && ok && attrsMap[export.JobNameAttribute] == "test-job" && attrsMap[export.UserNameAttribute] == "test-user"
 	})
-	suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), extraAttrsMatcher)
-	_, err = os.Stat("/tmp/scandata_export_100.csv")
+	suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, extraAttrsMatcher)
+	_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 	suite.Truef(os.IsNotExist(err), "Expected CSV file to be deleted")
 
 }
@@ -207,7 +210,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		repoCandidates := []int64{1}
 		artCandidates := []*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1, Digest: "digest1"}}}
@@ -236,7 +239,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 		err := suite.job.Run(ctx, params)
 		suite.NoError(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
 
@@ -248,8 +251,8 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 			_, ok := m[export.CreateTimestampKey]
 			return attrsMap[export.DigestKey] == MockDigest && ok
 		})
-		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), extraAttrsMatcher)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, extraAttrsMatcher)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		exportParamsMatcher := testifymock.MatchedBy(func(params export.Params) bool {
 			return reflect.DeepEqual(params.CVEIds, criteria.CVEIds)
@@ -271,7 +274,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		repoCandidate1 := &selector.Candidate{NamespaceID: 1}
 		repoCandidates := []*selector.Candidate{repoCandidate1}
@@ -298,7 +301,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 		err := suite.job.Run(ctx, params)
 		suite.NoError(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
 		m := make(map[string]interface{})
@@ -309,8 +312,8 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 			_, ok := m[export.CreateTimestampKey]
 			return attrsMap[export.DigestKey] == MockDigest && ok
 		})
-		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), extraAttrsMatcher)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, extraAttrsMatcher)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		exportParamsMatcher := testifymock.MatchedBy(func(params export.Params) bool {
 			return reflect.DeepEqual(params.CVEIds, criteria.CVEIds)
@@ -334,7 +337,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, errors.New("test error")).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
@@ -360,11 +363,11 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		err := suite.job.Run(ctx, params)
 		suite.Error(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-		suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, mock.Anything)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
 
@@ -384,7 +387,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{}, nil).Once()
@@ -410,11 +413,11 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		err := suite.job.Run(ctx, params)
 		suite.NoError(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, mock.Anything)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
 
@@ -436,7 +439,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(nil, errors.New("test error")).Once()
@@ -462,11 +465,11 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		err := suite.job.Run(ctx, params)
 		suite.Error(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-		suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, mock.Anything)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
 
@@ -486,7 +489,7 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		execAttrs := make(map[string]interface{})
 		execAttrs[export.JobNameAttribute] = "test-job"
 		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
+		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: ExecID, ExtraAttrs: execAttrs}, nil).Once()
 
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(nil, nil).Once()
@@ -512,11 +515,11 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		err := suite.job.Run(ctx, params)
 		suite.NoError(err)
 		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+			return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 		})
 		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
+		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, ExecID, mock.Anything)
+		_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 
 		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
 
@@ -538,11 +541,11 @@ func (suite *ScanDataExportJobTestSuite) TestExportDigestCalculationErrorsOut() 
 	err := suite.job.Run(ctx, params)
 	suite.Error(err)
 	sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-		return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
+		return sa.Repository == "scandata_export_1000000" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
 	})
 	suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
 	suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs")
-	_, err = os.Stat("/tmp/scandata_export_100.csv")
+	_, err = os.Stat("/tmp/scandata_export_1000000.csv")
 	suite.Truef(os.IsNotExist(err), "Expected CSV file to be deleted")
 }
 

--- a/src/pkg/scan/export/manager.go
+++ b/src/pkg/scan/export/manager.go
@@ -56,6 +56,8 @@ group by
 	`
 	JobModeExport = "export"
 	JobModeKey    = "mode"
+	JobID         = "JobId"
+	JobRequest    = "Request"
 )
 
 var (

